pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/TopicResources.java

```diff
@@ -127,10 +125,14 @@ public CompletableFuture<Void> clearTenantPersistence(String tenant) {
     }
 
     void handleNotification(Notification notification) {
+        if (topicListeners.isEmpty()) {
+            return;
+        }
         if (notification.getPath().startsWith(MANAGED_LEDGER_PATH)
```

Comment:
I think we can merge the early return guard into the `if` block like:

```java
        if (!topicListeners.isEmpty()
                && notification.getPath().startsWith(MANAGED_LEDGER_PATH)
                && (notification.getType() == NotificationType.Created
                || notification.getType() == NotificationType.Deleted)) {
```

It will be more consistent. Or you can use a merged guard like:

```java
        if (topicListeners.isEmpty()
                || !notification.getPath().startsWith(MANAGED_LEDGER_PATH)
                || (notification.getType() != NotificationType.Created
                && notification.getType() != NotificationType.Deleted)) {
            return;
        }
```

But it will bring more code changes.

Reply:
Thanks for the suggestion. Since it's not critical, I'll just keep it as-is to get this PR processed. :)